src/controllers/restatedeployment/reconcilers/knative.rs

```diff
@@ -0,0 +1,938 @@
+        }
+    }
+
+    // Build endpoint URL from Route default URL
+    let url = route
+        .status
+        .as_ref()
+        .and_then(|s| s.url.as_ref())
+        .ok_or_else(|| Error::RouteNotReady {
+            message: format!(
+                "Route {} does not have URL in status",
+                route.name_any()
+            ),
+            reason: "RouteURLNotReady".into(),
+            requeue_after: Some(Duration::from_secs(5)),
+        })?;
+
+    info!(url = %url, "Registering new deployment with Restate");
+
+    // Get admin URL
+    let admin_url = rsd.spec.restate.register.admin_url(&ctx.rce_store)?;
+
+    // Get bearer token if needed
+    let bearer_token = rsd.spec.restate.register.bearer_token(
+        &ctx.rce_store,
+        &ctx.secret_store,
+        &ctx.operator_namespace,
+    )?;
+
+    // Build registration request
+    let mut request = ctx
+        .http_client
+        .request(reqwest::Method::POST, admin_url.join("/deployments")?)
+        .json(&json!({
+            "uri": url,
+            "use_http_11": rsd.spec.restate.use_http11.unwrap_or(false),
+        }));
+
+    if let Some(token) = bearer_token {
+        request = request.bearer_auth(token);
+    }
+
+    let response = request.send().await.map_err(Error::AdminCallFailed)?;
+
+    if !response.status().is_success() {
+        let status = response.status();
+        let body = response.text().await.map_err(Error::AdminCallFailed)?;
+        return Err(Error::RestateRegistrationFailed(format!(
+            "Failed to register deployment: {} - {}",
+            status, body
+        )));
+    }
+
+    let response_json: serde_json::Value = response.json().await.map_err(Error::AdminCallFailed)?;
+    let deployment_id = response_json
+        .get("id")
+        .and_then(|v| v.as_str())
+        .ok_or_else(|| Error::RestateRegistrationFailed("No deployment ID in response".into()))?
+        .to_string();
+
+    info!(
+        deployment_id = %deployment_id,
+        "Successfully registered deployment"
+    );
+
+    Ok(deployment_id)
+}
+
+/// Annotate Configuration with deployment metadata
+async fn annotate_configuration(
+    ctx: &Context,
+    namespace: &str,
+    config: &Configuration,
+    deployment_id: &str,
+    tag: &str,
+) -> Result<()> {
+    let config_name = config.name_any();
+
+    let config_api: Api<Configuration> = Api::namespaced(ctx.client.clone(), namespace);
+    let params = PatchParams::apply("restate-operator/deployment-registration").force();
+
+    config_api
+        .patch_metadata(
+            &config_name,
+            &params,
+            &Patch::Apply(json!({
+                "apiVersion": "serving.knative.dev/v1",
+                "kind": "Configuration",
+                "metadata": {
+                    "annotations": {
+                        RESTATE_DEPLOYMENT_ID_ANNOTATION: deployment_id,
+                        RESTATE_TAG_ANNOTATION: tag,
```

Comment:
can't tag be set at config creation time? not sure why it has to come after registration

and re registered at annotation; do we need it? the mutate time will be in managed fields